qutebrowser/browser/pdfjs.py

```diff
@@ -0,0 +1,151 @@
+# vim: ft=python fileencoding=utf-8 sts=4 sw=4 et:
+
+# Copyright 2015 Daniel Schadt
+#
+# This file is part of qutebrowser.
+#
+# qutebrowser is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# qutebrowser is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with qutebrowser.  If not, see <http://www.gnu.org/licenses/>.
+
+"""pdf.js integration for qutebrowser."""
+
+import os
+
+from qutebrowser.browser import webelem
+from qutebrowser.utils import utils
+
+
+class PDFJSNotFound(Exception):
+
+    """Raised when no pdf.js installation is found."""
+
+    pass
+
+
+def generate_pdfjs_page(url):
+    """Return the html content of a page that displays url with pdfjs.
+
+    Returns a string.
+
+    Args:
+        url: The url of the pdf as QUrl.
+    """
+    viewer = get_pdfjs_res('web/viewer.html').decode('utf-8')
+    script = _generate_pdfjs_script(url)
+    html_page = viewer.replace(
+        '</body>', '</body><script>{}</script>'.format(script)
+    )
+    return html_page
+
+
+def _generate_pdfjs_script(url):
+    """Generate the script that shows the pdf with pdf.js.
+
+    Args:
+        url: The url of the pdf page as QUrl.
+    """
+    return (
+        'PDFJS.getDocument("{url}").then(function(pdf) {{\n'
+        '    PDFView.load(pdf);\n'
+        '}});'
+    ).format(url=webelem.javascript_escape(url.toString()))
```

Comment:
I think you should pass `QUrl.FullyEncoded` to `toString` here, otherwise this might break with funny URLs (like with spaces or non-ascii chars). Ideally there'd be some tests for that of course :wink: